### PR TITLE
Features/pull impl

### PR DESCRIPTION
**Pull request by bygui86**
This is the first PR from this branch.
First of all I wanted to understand the codebase.
Then I proceeded for a refactoring to better respect best practices in terms of methods, packages, responsibility segregation and comments.
In the next PRs I will implement the `pull` feature.

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package utils
```
Comment (mkretz) [inline utils/utils.go]: I don't agree with having a package named "utils". This is for sure not best practice. Why are those functions in this package and not others? Why are they public

File: main.go [hunk 2]
```diff
@@ -1,98 +1,148 @@
...
 
+const (
+	version = "0.0.1"
+)
+
+// main -
 func main() {
-	var scanpath, registry, output string
+
+	setLogLevel("debug")
```
Comment (mkretz) [inline main.go]: OK sure, we can do more elaborate logging, but then we should give the user an option to switch it on and off when running korp.

File: utils/utils.go (hunk 1 above)
Comment (mkretz) [inline utils/utils.go]: I don't agree with having a package named "utils". This is for sure not best practice. What's the responsibility of this package? Why are those functions in this package and not others? Why should these functions be public in the whole app? We only need them when parsing the yml-files. They belong in the actions package as private members in my opinion.

File: main.go [hunk 3]
```diff
@@ -10,19 +10,21 @@ import (
 )
 
 const (
-	version = "0.0.1"
+	version  = "0.0.1"
+	logLevel = "debug"
```
Comment (mkretz) [inline main.go]: See previous comment. Can we make this an option for the user? Should be simple enough using the CLI library.

File: main.go (hunk 2 above)
Comment (bygui86) [reply main.go]: Sure, we can insert it in the todo list and create another specific feature branch later :)

File: utils/utils.go (hunk 1 above)
Comment (bygui86) [reply utils/utils.go]: You are right, the naming convention is not the best.
This package includes all utilities we are going to need around other packages.
At the end I will split this package into small more focused ones.

File: main.go (hunk 3 above)
Comment (bygui86) [reply main.go]: The log level sure, the version is for us ;) and the use of go-releaser

**Pull request merged**